### DISCUSSION TITLE
실습코드 5-5 오류 with 스프링부트 2.6이상버전

### DISCUSSION THREAD

**Discussion by bluejick** [fsoftwareengineer/todo-application#41] [질문과 답변] [answered]
개발환경을 스프링부트2.6.2기반으로 구성하여 진행하던 중..
실습코드 5-5에서 오류를 맞이하였습니다.
이것저것 해보다가 스프링부트버전만 2.5.8로 낮추니 잘 되더군요..
오류는 아래와 같은데.. 
짧은 지식으로 생각해보면 에러 응답이 JSON형태가 아닌거 같은데요..

```
GET http://localhost:8080/todo 403
ApiService.js:27 SyntaxError: Unexpected end of JSON input at ApiService.js:15
VM10:1 Uncaught (in promise) SyntaxError: Unexpected end of JSON input at ApiService.js:15
```

실습이야 그냥 2.5.8로 낮춰서 진행하면 된다지만..
 
2.6에서 어떤 점 때문에 오류가 나는건지.. 
이걸 스프링 2.6 이상버전 자체의 오류라고 해야 할지..
아니면 무언가가 바뀌었고 해결을 위해 어떤 다른 설정이 필요한건지.. 
궁금해서 질문 남깁니다.

테스트는 
스프링부트 2.5.0-SNAPSHOT => 정상
스프링부트 2.5.8 => 정상
스프링부트 2.5.9-SNAPSHOT => 정상
스프링부트 2.6.1 => 오류
스프링부트 2.6.2 => 오류
스프링부트 2.6.3-SNAPSHOT => 오류
스프링부트 2.7.0-SNAPSHOT => 오류

**Comment by fsoftwareengineer** [comment]
확인하는 방법으로는 포스트맨을 이용해 같은 api를 버전이 다른 환경에서 호출 한 후 결과를 비교하는 방법이 있습니다.

**Reply by bluejick** [reply]
감사합니다. 
조언을 따라 포스트맨을 이용해 호출해 보니 2.6버전이상부터 에러 응답이 비어있는것을 확인했습니다. 
관련해서 검색을 해보니.. 해당문제는 스프링부트2.6버전의 자체 오류였으나 2.6.2버전에서 해결되었다는 글을 발견하였습니다.
https://www.javacodexamples.com/examples/error-response-body-is-empty-in-spring-boot-2-6

그러나.. 저의 경우 2.6.2버전을 사용해도 계속해서 비어있는 응답이 들어오네요.. ㅠㅠ
혹시나 싶어 server.error 프로퍼티를 바꿔보기도 했는데요.. 결과는 동일합니다. 

일단.. 해당 문제는 어쨌든 스프링부트 자체 문제인 것 같습니다.

실습은 그냥 2.5.8버전으로 계속 진행하도록 하겠습니다~
 
답변 감사합니다~!

**Reply by bluejick** [reply]
아래 @Senna-Lab 님의 조언대로 진행하니 잘 됩니다~! ^^

**Comment by Senna-Lab** [comment, marked as answer]
저도 같은 문제로 고생했었습니다.

검색으로 조사하니,
Spring boot 2.6 부터 'an unauthenticated user no longer receives a full error response.' 라고 하는 것 같네요.
https://github.com/spring-projects/spring-boot/issues/28953

그러므로 Spring Boot 2.6.~ 사용하는 경우에 인증되지 않은 유저에게도 error response를 보내려면,
WebSecurityConfig.java에서 "/error"에 대한 접근권한을 설정해줘야 하는 것 같습니다.
![image](https://user-images.githubusercontent.com/57217392/147806400-b398f688-3668-4457-911c-9fcaff1c11ce.png)

**Reply by bluejick** [reply]
감사합니다~! "/error" 접근권한을 주니 잘 되네요~! ^^
